### PR TITLE
feat: added content for Typography component for doc site

### PR DESCRIPTION
**Pull request by zero0Halo**
feat: added readme text

chore: added details for typography's props

chore: Added a section for default styles

chore: started making the props section look more like autocomplete's api section

chore: made the props section the api section, all bulleted props are now in a table

**Before submitting a pull request,** please make sure the following is done:

I have done **all** of the following:

- [ ] Added a top level class to all my components `'.anchor-[COMPONENT NAME]'`.
- [x] Used [conventional commits](https://www.conventionalcommits.org) for all work.
- [ ] Tested my solution on Mobile & Tablet.
- [ ] Wrote [unit tests](https://jestjs.io/docs/en/getting-started) for states and all behavior (`npm test`) and passed coverage thresholds.
- [x] Updated snapshots for all permutations (`npm test -- -u`).
- [ ] Accounted for hover, focus, blur, visited, & error states because they are not edge cases.
- [ ] Created TODOs for known edge cases.
- [x] Documented all of my changes (inline & doc site).
- [ ] Made sure that all accessibility errors are resolved.
- [ ] Added [stories](https://storybook.js.org/docs/basics/introduction/) with knobs for all possible configurations.
- [x] De-linted and ran [prettier](https://github.com/prettier/prettier) (`npm run pretty`) on my code.
- [ ] Added name to OWNERS file for all new components
- [ ] If adding a new component, add its export to the rollup config

---------
**Outline your feature or bug-fix below**

This adds documentation for the Typography component in the doc site. In addition, it's my first attempt at documenting a component that I did not myself build. Comments are very much encouraged on the accuracy of my understanding.

### CODE REVIEW

File: docs/src/pages/components/typography.mdx [hunk 1]
```diff
@@ -1 +1,186 @@
...
+* `font-family: Avenir Next, Segoe UI, Helvetica Neue, Helvetica, Roboto,sans-serif;`
+* `font-size: 1rem;`
+* `line-height: 1.5rem;`
+* `text-align: inherit;`
+* `text-transform: none;`
+* `color: inherit;`
+
+<br />
+
+#### Inline Code Editor
```
Comment (sharkparty) [inline docs/src/pages/components/typography.mdx:28]: We should add a descriptor in here about the reason we have this notion of `scale`. It's (1) our way of mitigating the fact that semantic HTML tagging doesn't always get the same text styles and we needed a mechanism for handling that and (2) we needed a consistent way to render common fonts regardless of their enclosing tags for CTAs.

**Review by sharkparty — CHANGES_REQUESTED** [review]
Adding an addendum for `scale` vs `size`/`weight`/`line-height`

File: docs/src/pages/components/typography.mdx (hunk 1 above)
Comment (zero0Halo) [reply docs/src/pages/components/typography.mdx]: Updated description: "This value directly affects the size and lineHeight props directly, ensuring
                that there is no overlap of text between lines no matter what value is input.
                Often, semantic HTML tagging doesn't always get the same text styles and this helps
                alleviate that problem. It also assists in rendering common fonts regardless of 
                their enclosing tags."

Comment (zero0Halo) [reply docs/src/pages/components/typography.mdx:28]: I've added a paragraph to this effect.

**Review by sharkparty — APPROVED** [review]
LGTM

**Pull request merged**